src/Core/Migration/V6_7/Migration1728040169AddPrimaryOrderDeliveryAndTransaction.php

```diff
@@ -0,0 +1,77 @@
+        // address.
+
+        if (!$this->columnExists($connection, 'order', 'primary_order_delivery_id')) {
+            $connection->executeStatement(
+                'ALTER TABLE `order`
+                ADD COLUMN `primary_order_delivery_id` BINARY(16) NULL DEFAULT NULL,
+                ADD COLUMN `primary_order_delivery_version_id` BINARY(16) NULL DEFAULT NULL,
+                ADD UNIQUE INDEX `uidx.order.primary_order_delivery` (`id`, `version_id`, `primary_order_delivery_id`);'
+            );
+
+            $connection->executeStatement(
+                'UPDATE `order`
+                INNER JOIN `order_delivery` as `primary_order_delivery`
+                    ON `primary_order_delivery`.`order_id` = `order`.`id`
+                    AND `primary_order_delivery`.`order_version_id` = `order`.`version_id`
+                    AND `primary_order_delivery`.`id` = (
+                        SELECT `id`
+                        FROM `order_delivery`
+                        WHERE `order_delivery`.`order_id` = `order`.`id`
+                        AND `order_delivery`.`order_version_id` = `order`.`version_id`
+                        ORDER BY `order_delivery`.`created_at` DESC
+                        LIMIT 1
+                    )
+                SET `order`.`primary_order_delivery_id` = `primary_order_delivery`.`id`
+                WHERE `order`.`primary_order_delivery_id` IS NULL;'
+            );
+        }
+
+        if (!$this->columnExists($connection, 'order', 'primary_order_transaction_id')) {
+            $connection->executeStatement(
+                'ALTER TABLE `order`
+                ADD COLUMN `primary_order_transaction_id` BINARY(16) NULL DEFAULT NULL,
+                ADD COLUMN `primary_order_transaction_version_id` BINARY(16) NULL DEFAULT NULL,
+                ADD UNIQUE INDEX `uidx.order.primary_order_transaction` (`id`, `version_id`, `primary_order_transaction_id`);'
+            );
+            $connection->executeStatement(
+                'UPDATE `order`
+                INNER JOIN `order_transaction` as `primary_order_transaction`
+                    ON `primary_order_transaction`.`order_id` = `order`.`id`
+                    AND `primary_order_transaction`.`order_version_id` = `order`.`version_id`
+                    AND `primary_order_transaction`.`id` = (
+                        SELECT `id`
+                        FROM `order_transaction`
+                        WHERE `order_transaction`.`order_id` = `order`.`id`
+                        AND `order_transaction`.`order_version_id` = `order`.`version_id`
+                        ORDER BY `order_transaction`.`created_at` DESC
+                        LIMIT 1
+                    )
+                SET `order`.`primary_order_transaction_id` = `primary_order_transaction`.`id`
+                WHERE `order`.`primary_order_transaction_id` IS NULL;'
+            );
+        }
+    }
```

Comment:
can we split the two columns into two Migrations? I think, this makes more sense for large DBs.